crates/core/tedge/src/main.rs

```diff
@@ -120,20 +124,39 @@ fn executable_name() -> Option<String> {
     )
 }
 
-fn parse_multicall_if_known<T: Parser>(executable_name: &Option<String>) -> T {
-    let cmd = T::command();
+fn parse_multicall_if_known(executable_name: &Option<String>) -> TEdgeOptMulticall {
+    let cmd = TEdgeOptMulticall::command();
 
     let is_known_subcommand = executable_name
         .as_deref()
         .map_or(false, |name| cmd.find_subcommand(name).is_some());
     let cmd = cmd.multicall(is_known_subcommand);
 
     let cmd2 = cmd.clone();
-    match T::from_arg_matches(&cmd.get_matches()) {
+    match TEdgeOptMulticall::from_arg_matches(&cmd.get_matches()) {
+        Ok(TEdgeOptMulticall::Tedge { cmd, common }) => redirect_if_multicall(cmd, common),
         Ok(t) => t,
         Err(e) => {
             eprintln!("{}", RichFormatter::format_error(&e.with_cmd(&cmd2)));
             std::process::exit(1);
         }
     }
 }
+
+// Transform `tedge mapper|agent|write` commands into multicalls
+//
+// This method has to be kept in sync with TEdgeOpt::build_command
+fn redirect_if_multicall(cmd: TEdgeOpt, common: CommonArgs) -> TEdgeOptMulticall {
+    match cmd {
+        TEdgeOpt::Mapper(opt) => {
+            TEdgeOptMulticall::Component(Component::TedgeMapper(opt.with_common_args(common)))
```

Comment:
We have to be more cautious here: there is a risk of confusion as `--tedge-config` is provided twice to `clap`.

Reply:
I assume you mean `--config-dir`

Reply:
When a given global argument (as `--config-dir`) is used twice by the command struct (as here where `common` can be accessed directly (`common`) or via the mapper options (`opt.common`), `clap` correctly duplicate the user provided info to all the inner components.

- Hence, there is no need to explicitly copy to common arg.
- As this clap behavior is not documented, tests have been added, see https://github.com/thin-edge/thin-edge.io/pull/3313/commits/b7353194746dbe0a86ab0a17cd48cd7aaafa5975
- There is one caveat: `clap` fails to notice that an argument is provided twice.